test/composer.test.ts

```diff
@@ -23,48 +54,435 @@ describe("Composer", () => {
 
     it("should call handlers", async () => {
         composer.use(middleware);
-        await composer.middleware()(ctx, next);
+        await exec();
```

Comment:
What’s the benefit of hiding this in the `exec` function?

Reply:
I expected `exec` to be rather complex so I did not want to repeat myself. But it's also kinda opaque, so I'm not happy either way. Suggestions? Inline `exec` everywhere?

Reply:
Is that upvote a yes to the inlining?